### DISCUSSION TITLE
Using Fluture with Jest

### DISCUSSION THREAD

**Discussion by kee-oth** [fluture-js/Fluture#491] [Q&A]
Hi!

I'm just getting into using Fluture and I was wondering if there is a recommended way to use Fluture with Jest? The issue is that Jest's "expect" function needs to be able to throw an error but Fluture consumes them.

Here's my code. Even if `expect` fails here Jest reports that the test passes.

```
  const findByRole = param1 => encaseP (param2 => (
    screen.findByRole (param1, param2)
  ))

  findByRole ('button') ({ name: 'Click me!' })
    .pipe (
      fork (handleRejection) (component => {
        expect (component).toBeInTheDocument ()
        fireEvent.click (component)
      }),
    )
```

Thanks!

**Comment by dicearr** [comment]
You could use the [`equivalence`](https://github.com/fluture-js/Fluture/blob/4667b944fa1fabe9d0a7fe7d9e6c57865ffd5be2/test/assertions.js#L115) function from `fluture/test/assertions.js`.

```js
import {equivalence} from 'fluture/test/assertions.js';
import {encaseP, map, resolve} from 'fluture';

const findByRole = param1 => encaseP (param2 => screen.findByRole (param1, param2));

findByRole ('button') ({name: 'Click me!'})
.pipe (map (component => expect(component).toBeInTheDocument ()))
.pipe (equivalence (resolve (/* Whatever toBeInTheDocument returns */)));
```

But I would probably implement an `assertResolved` function. Similar to the `makeEquivalence` function exported in `fluture/test/assertions.js`. So you can ignore the inner value within the Future in your tests.
